contracts/bridge/L1ECOBridge.sol

```diff
@@ -48,10 +55,13 @@ contract L1ECOBridge is IL1ECOBridge, CrossDomainEnabledUpgradeable {
      */
     modifier onlyEOA() {
         // Used to stop deposits from contracts (avoid accidentally lost tokens)
-        require(!Address.isContract(msg.sender), "Account not EOA");
+        require(msg.sender.code.length == 0, "Account not EOA");
```

Comment:
not sure how much it matters for this case but this check can apparently be circumvented by making this call in the constructor of a contract, at which point the bytecode at the contract address is still 0. regardless of whether thats relevant here it might come up in audits. can be remedied by using the address library or I think checking that msg.sender == tx.origin should also work

Reply:
not only is this the exact code from the address library, but also your concern is already mentioned in the comment above this modifier

Reply:
ah gotcha my b